btf/btf_test.go

```diff
@@ -7,12 +7,38 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"sync"
 	"testing"
 
 	"github.com/cilium/ebpf/internal"
 	"github.com/cilium/ebpf/internal/testutils"
 )
 
+// vmlinux caches the result of parsing the running kernel's BTF.
+var vmlinux struct {
+	sync.Once
+	spec *Spec
+	err  error
+}
+
+func vmlinuxSpec(tb testing.TB) *Spec {
+	tb.Helper()
+
+	vmlinux.Do(func() {
+		vmlinux.spec, vmlinux.err = LoadKernelSpec()
+	})
+	err := vmlinux.err
+
+	if errors.Is(err, ErrNotSupported) {
```

Comment:
Why not `SkipIfNotSupported`?

I'm a bit torn about skipping on ErrNotSupported here, I think it's better to hardcode `SkipOnOldKernel` to coincide with the availability of `/sys/fs/...`. ErrNotSupported is convenient but makes it too easy to skip tests in subtle ways.

Reply:
> Why not SkipIfNotSupported?
Good catch!

>  I think it's better to hardcode SkipOnOldKernel to coincide with the availability of /sys/fs/...

Agreed, this shouldn't cause any problems running on dev machines, and in CI we only run upstream kernels so I guess we can make a hard assumption here.

Reply:
Done, also made `TestParseCurrentKernelBTF` use the cached `vmlinux`.